packages/core/test/scuttle.spec.js

```diff
@@ -0,0 +1,126 @@
+const test = require('ava')
+const { evaluateWithSourceUrl } = require('./util')
+const { scuttle } = require('../src/scuttle')
+
+const SCUTTLER_NAME_ERROR = {
+  message:
+    'LavaMoat - \'scuttlerName\' function "SCUTTLER" expected on globalRef.' +
+    'To learn more visit https://github.com/LavaMoat/LavaMoat/pull/462.',
+}
+
+const err = (intrinsic) =>
+  'LavaMoat - property "' +
+  intrinsic +
+  '" of globalThis is inaccessible under ' +
+  'scuttling mode. To learn more visit https://github.com/LavaMoat/LavaMoat/pull/360.'
+
+function testScuttling(
+  globalRef,
+  intrinsics = ['Array', 'Int8Array', 'RegExp']
+) {
+  return intrinsics.some((intrinsic) => {
+    try {
```

Comment:
why do we look for an exception here? the purpose of this function isn't very clear to me

Reply:
Scuttling means you replace the getter of a property on the global object with a getter that throws instead, so that access attempt to a forbidden property will throw an exception for users to easily tell what went wrong.
So to test that, we need to prove expected exceptions are in fact thrown.
If however the exception caught in `testScuttling` is not what we expect, I want the test to break.
`testScuttling`'s purpose is to return a bool of whether a prop is scuttled or not.

Reply:
> Scuttling means you replace the getter of a property on the global object with a getter that throws instead,

I am curious about why this is, when we use no-ops elsewhere, right?

Reply:
> If however the exception caught in testScuttling is not what we expect, I want the test to break.

Why don't we use `ava`'s own facilities to do this?  `t.throws`

Reply:
> I am curious about why this is, when we use no-ops elsewhere, right?
https://github.com/LavaMoat/LavaMoat/pull/360#discussion_r946406834

Reply:
> Why don't we use ava's own facilities to do this? t.throws
fixed 6800efdaf